File: frontend/src/api/schemas.ts
export interface ProfileOut {
  id: number;
  email: string;
  avatar?: string;
  first_name?: string;
  last_name?: string;
  phone_number?: string;
  date_of_birth?: string;
  height?: number;
  weight?: number;
  chest?: number;
  waist?: number;
  hips?: number;
  favorite_colors?: string | string[];
  favorite_brands?: string | string[];
  is_admin?: boolean;
}

export interface TokensUserOut {
  access_token: string;
  token_type: string;
  refresh_token: string;
  user: ProfileOut;
}

export interface UserOut {
  id: number;
  email: string;
  is_admin: boolean;
  is_active: boolean;
}

export interface UserCreateAdmin {
  email: string;
  password: string;
  is_admin?: boolean;
  is_active?: boolean;
}

export interface UserUpdateAdmin {
  email?: string;
  password?: string;
  is_admin?: boolean;
  is_active?: boolean;
}

export interface VariantOut {
  size?: string;
  color?: string;
  sku?: string;
  stock?: number;
  price?: number;
  id: number;
}

export interface ItemOut {
  name: string;
  brand?: string;
  color?: string;
  image_url?: string;
  description?: string;
  price?: number;
  category?: string;
  clothing_type?: string;
  article?: string;
  size?: string;
  style?: string;
  collection?: string;
  id: number;
  created_at?: string;
  updated_at?: string;
  image_urls?: string[];
  variants?: VariantOut[];
  is_favorite?: boolean;
}

export interface OutfitItemBase {
  id: number;
  name: string;
  brand?: string;
  image_url?: string;
  price?: number;
}

export interface OutfitOut {
  id: number;
  name: string;
  style: string;
  description?: string;
  collection?: string;
  owner_id: string;
  created_at?: string;
  updated_at?: string;
  tops?: OutfitItemBase[];
  bottoms?: OutfitItemBase[];
  footwear?: OutfitItemBase[];
  accessories?: OutfitItemBase[];
  fragrances?: OutfitItemBase[];
  total_price?: number;
}

export interface ProfileUpdate {
  avatar?: string;
  first_name?: string;
  last_name?: string;
  phone_number?: string;
  date_of_birth?: string;
  height?: number;
  weight?: number;
  chest?: number;
  waist?: number;
  hips?: number;
  favorite_colors?: string | string[];
  favorite_brands?: string | string[];
}

export interface CartItemOut {
  item_id: number;
  variant_id: number;
  name: string;
  brand?: string;
  image_url?: string;
  size?: string;
  color?: string;
  sku?: string;
  quantity: number;
  price?: number;
  stock?: number;
}

export interface CartStateOut {
  items: CartItemOut[];
  total_items: number;
  total_price: number;
}

export interface OutfitCreate {
  name: string;
  style: string;
  description?: string;
  collection?: string;
  top_ids?: number[];
  bottom_ids?: number[];
  footwear_ids?: number[];
  accessories_ids?: number[];
  fragrances_ids?: number[];
}

export interface OutfitUpdate {
  name?: string;
  style?: string;
  description?: string;
  collection?: string;
  top_ids?: number[];
  bottom_ids?: number[];
  footwear_ids?: number[];
  accessories_ids?: number[];
  fragrances_ids?: number[];
}

export interface OutfitCommentCreate {
  content: string;
  rating?: number;
}

export interface OutfitCommentOut {
  content: string;
  rating?: number;
  id: number;
  user_id: number;
  created_at: string;
  likes: number;
}

export interface ItemUpdate {
  name?: string;
  brand?: string;
  color?: string;
  image_url?: string;
  description?: string;
  price?: number;
  category?: string;
  clothing_type?: string;
  article?: string;
  size?: string;
  style?: string;
  collection?: string;
}

export interface CommentCreate {
  content: string;
  rating?: number;
}

export interface CommentOut {
  content: string;
  rating?: number;
  id: number;
  user_id: number;
  user_name?: string;
  created_at: string;
  likes: number;
}

export interface VariantCreate {
  size?: string;
  color?: string;
  sku?: string;
  stock?: number;
  price?: number;
}

export interface VariantUpdate {
  size?: string;
  color?: string;
  sku?: string;
  stock?: number;
  price?: number;
}

export interface Body_create_item_api_items__post {
  name: string;
  brand?: string;
  color?: string;
  description?: string;
  price?: number;
  category?: string;
  article?: string;
  size?: string;
  style?: string;
  collection?: string;
  images?: File[];
  image_url?: string;
  clothing_type?: string;
} 

File: backend/app/api/v1/endpoints/items/schemas.py
from typing import List, Optional
from pydantic import BaseModel, conint, confloat
from datetime import datetime


class ItemCreate(BaseModel):
    name: str
    brand: Optional[str] = None
    color: Optional[str] = None
    image_url: Optional[str] = None
    description: Optional[str] = None
    price: Optional[confloat(gt=0)] = None
    category: Optional[str] = None
    clothing_type: Optional[str] = None
    article: Optional[str] = None
    size: Optional[str] = None
    style: Optional[str] = None
    collection: Optional[str] = None


class ItemUpdate(BaseModel):
    name: Optional[str] = None
    brand: Optional[str] = None
    color: Optional[str] = None
    image_url: Optional[str] = None
    description: Optional[str] = None
    price: Optional[confloat(gt=0)] = None
    category: Optional[str] = None
    clothing_type: Optional[str] = None
    article: Optional[str] = None
    size: Optional[str] = None
    style: Optional[str] = None
    collection: Optional[str] = None


class VariantBase(BaseModel):
    size: Optional[str] = None
    color: Optional[str] = None
    sku: Optional[str] = None
    stock: conint(ge=0) = 0
    price: Optional[confloat(gt=0)] = None


class VariantCreate(VariantBase):
    pass


class VariantUpdate(BaseModel):
    size: Optional[str] = None
    color: Optional[str] = None
    sku: Optional[str] = None
    stock: Optional[conint(ge=0)] = None
    price: Optional[confloat(gt=0)] = None


class VariantOut(VariantBase):
    id: int

    class Config:
        orm_mode = True


class ItemOut(ItemCreate):
    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    style: Optional[str] = None
    collection: Optional[str] = None
    image_urls: Optional[List[str]] = None
    variants: Optional[List[VariantOut]] = None
    is_favorite: Optional[bool] = None
    
    # Новые поля для Lamoda
    source: Optional[str] = None
    source_url: Optional[str] = None
    source_sku: Optional[str] = None
    old_price: Optional[float] = None

    class Config:
        orm_mode = True


class CommentCreate(BaseModel):
    content: str
    rating: Optional[conint(ge=1, le=5)] = None  # 1-5


class CommentOut(CommentCreate):
    id: int
    user_id: int
    user_name: Optional[str] = None
    created_at: Optional[datetime]
    likes: Optional[int] = 0

    class Config:
        orm_mode = True


# Images


class ItemImageOut(BaseModel):
    id: int
    image_url: str
    position: Optional[int] = None

    class Config:
        orm_mode = True


# Lamoda Import


class LamodaImportRequest(BaseModel):
    query: str
    limit: Optional[int] = 20
    domain: Optional[str] = "kz"
    
    class Config:
        schema_extra = {
            "example": {
                "query": "nike кроссовки",
                "limit": 10,
                "domain": "kz"
            }
        }


class LamodaImportResponse(BaseModel):
    success: bool
    message: str
    imported_count: int
    items: List[ItemOut]
    
    class Config:
        schema_extra = {
            "example": {
                "success": True,
                "message": "Successfully imported 5 items from Lamoda",
                "imported_count": 5,
                "items": []
            }
        } 